### PR TITLE
Sql error fix

### PR DESCRIPTION
**Pull request by SamTov**
Fix issue #223 for Green-Kubo and Einstein distinct diffusion coefficients.

### CODE REVIEW

File: mdsuite/calculators/einstein_distinct_diffusion_coefficients.py [hunk 1]
```diff
@@ -230,29 +230,26 @@ def _post_operation_processes(self, species: Union[str, tuple] = None):
             result = self._fit_einstein_curve([self.time, abs(self.msd_array)])
             properties = {"Property": self.database_group,
                           "Analysis": self.analysis_name,
-                          "Subject": "_".join(species),
+                          "Subject": [species[0], species[1]],
```
Comment (PythonFZ) [inline mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:233]: `"Subject":species` should also work then, I think?

File: mdsuite/calculators/einstein_distinct_diffusion_coefficients.py [hunk 2]
```diff
@@ -230,29 +230,26 @@ def _post_operation_processes(self, species: Union[str, tuple] = None):
...
-                          'uncertainty': result[1]}
+                          'data': [{'x': -1*result[0], 'uncertainty': result[1]}]
+                          }
             self._update_properties_file(properties)
         else:
             result = self._fit_einstein_curve([self.time, self.msd_array])
             properties = {"Property": self.database_group,
                           "Analysis": self.analysis_name,
-                          "Subject": species,
+                          "Subject": [species[0], species[1]],
```
Comment (PythonFZ) [inline mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:242]: see above

File: mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py [hunk 3]
```diff
@@ -232,7 +232,7 @@ def _post_operation_processes(self, species: Union[str, tuple] = None):
 
         properties = {"Property": self.database_group,
                       "Analysis": self.analysis_name,
-                      "Subject": species,
+                      "Subject": [species[0], species[1]],
```
Comment (PythonFZ) [inline mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py:235]: Did this really fail?

File: mdsuite/calculators/einstein_distinct_diffusion_coefficients.py [hunk 4]
```diff
@@ -230,29 +230,26 @@ def _post_operation_processes(self, species: Union[str, tuple] = None):
...
+                          'data': [{'x': result[0], 'uncertainty': result[1]}]
+                          }
             self._update_properties_file(properties)
 
         # Update the plot if required
         if self.plot:
             plt.plot(np.array(self.time) * self.experiment.units['time'], self.msd_array, label=species)
             plt.show()
 
-        if self.save:
```
Comment (PythonFZ) [inline mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:253]: I think, you should be able to change this to

```py
if self.save:
    properties = {"Property": self.database_group,
                  "Analysis": self.analysis_name,
                  "Subject": species,
                  "data_range": self.data_range,
                  'data': [{'x': x, 'y': y} for x, y in zip(self.time, self.msd_array)],
                  'information': "series"}
    self._update_properties_file(properties)
```
But I am still unsure about the species

**Review by PythonFZ — CHANGES_REQUESTED** [review]
Have you tested the change:

```py
"Subject": species
"Subject": [species[0], species[1]]
```

It might be redundant.

File: mdsuite/calculators/einstein_distinct_diffusion_coefficients.py (hunk 1 above)
Comment (SamTov) [reply mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:233]: Species is a tuple but yes I am sure I could also do something like list(species)

Comment (SamTov) [reply mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:233]: Okay so yes one can just write list(species)

**Review by PythonFZ — APPROVED** [review]

File: mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py (hunk 3 above)
Comment (SamTov) [reply mdsuite/calculators/green_kubo_distinct_diffusion_coefficients.py]: I tried it with just the tuple and yes it failed.

**Comment by PythonFZ** [comment]
You may look into the save method and test it again, but I am confident, that everything should work fine.

File: mdsuite/calculators/einstein_distinct_diffusion_coefficients.py (hunk 4 above)
Comment (SamTov) [reply mdsuite/calculators/einstein_distinct_diffusion_coefficients.py:253]: The tuple in the 'Subject' throws and error. It needs to be a list, unless you have changed something else.

**Comment by SamTov** [comment]
I changed the species to list(species)

**Pull request merged**